contracts/Wrapped1155Factory.sol

```diff
@@ -135,16 +145,32 @@ contract Wrapped1155Factory is ERC1155Receiver {
             
             // assign name
             hex"7f",
-            "Wrapped ERC-1155", uint128(32),
+            // Chars in hex or UTF-8 = 8 bits => 2 hex numbers => 
+            // string + uint128(30) for 16 chars needs to padding zeros 
+            // and the length of the string
+            // leading zeros = (64 - (tokenName.length * 2)) / 2
+            // Wrapped ERC-1155GnosisProtocol = 31 * 2 zeros + length * 2 in hex
+            // (62 - (31 * 2)) = 0 zeros + length * 2 in hex
+            // "Wrapped ERC-1155", uint128(32), 
+            // Wrapped ERC-1155 = 16 zeros + length * 2 in hex
+            // WrappedERC-1155 = 15 * 2 zeros + length * 2 in hex
+            // (62 - (15 * 2)) = 16
+            //"WrappedERC-1155", hex"000000000000000000000000000000001E", 
+            tokenName, tokenNameOffset, 
```

Comment:
So the name `tokenNameOffset` is misleading in my opinion. As this is the padding of the name + the length.

Also I would link the solidity docs for more clearification: https://docs.soliditylang.org/en/v0.8.1/internals/layout_in_storage.html#bytes-and-string

and would add a comment that is similar to:
`tokenName` can be at max be 31bytes long (actually needs to be padded to 31bytes) and length should added after.

Looking at this I would actually recommend to not use `bytes` as type for `tokenName` and `tokenNameOffset`, but rather use `bytes31` for `tokenName` and `uint8` for `tokenNameLength` (currently `tokenNameOffset`). 

If you want to stick to the current types I would recommend a check that `tokenName.length + tokenNameOffset.length == 32`

Reply:
Thanks you @rmeissner : I will rename them and replace the types. I would also like to make it fit the parameters with the override methods using the calldata data parameter.

Reply:
Hi @rmeissner , I'm sorry I had to revert to bytes. What I need is pass the metadata in bytecode for the `getWrapped1155DeployBytecode`. 

The problem is that we can non add more parameter on override method from 1155 (`onERC1155Received` and `onERC1155BatchReceived`), so we are using the `bytes calldata data` parameter to send the token `name`, `symbol`, and `decimal` bytecode as an array of bytes. 

The `getWrapped1155DeployBytecode` will slice the bytes.

I included a `1155-to-20-helper` on the tests and that can be use by the client side.